ipatests/test_integration/test_ipahealthcheck.py

```diff
@@ -0,0 +1,507 @@
+        """
+        Testcase to verify sources available in healthcheck tool.
+        """
+        result = self.master.run_command(["ipa-healthcheck", "--list-sources"])
+        assert result.returncode == 0
+        for source in sources:
+            assert source in result.stdout_text
+
+    def test_dogtag_ca_check_exists(self):
+        """
+        Testcase to verify checks available in
+        ipahealthcheck.dogtag.ca source
+        """
+        result = self.master.run_command(
+            ["ipa-healthcheck", "--source", "ipahealthcheck.dogtag.ca"]
+        )
+        for check in dogtag_checks:
+            assert check in result.stdout_text
+
+    def test_replication_check_exists(self):
+        """
+        Testcase to verify checks available in
+        ipahealthcheck.ds.replication source
+        """
+        result = self.master.run_command(
+            ["ipa-healthcheck", "--source", "ipahealthcheck.ds.replication"]
+        )
+        for check in replication_checks:
+            assert check in result.stdout_text
+
+    def test_ipa_cert_check_exists(self):
+        """
+        Testcase to verify checks available in
+        ipahealthcheck.ipa.certs source
+        """
+        result = self.master.run_command(
+            ["ipa-healthcheck", "--source", "ipahealthcheck.ipa.certs"]
+        )
+        for check in ipa_cert_checks:
+            assert check in result.stdout_text
+
+    def test_ipa_trust_check_exists(self):
+        """
+        Testcase to verify checks available in
+        ipahealthcheck.ipa.trust source
+        """
+        result = self.master.run_command(
+            ["ipa-healthcheck", "--source", "ipahealthcheck.ipa.trust"]
+        )
+        for check in ipatrust_checks:
+            assert check in result.stdout_text
+
+    def test_source_ipahealthcheck_meta_services_check_sssd(self):
+        """
+        Testcase checks behaviour of check sssd in
+        ipahealthcheck.meta.services when service is stopped and started
+        respectively
+        """
+        cmd = self.master.run_command(["systemctl", "stop", "sssd"])
+        assert cmd.returncode == 0
+        self.master.run_command(
+            [
+                "ipa-healthcheck",
+                "--source",
+                "ipahealthcheck.meta.services",
+                "--check",
+                "sssd",
+                "--output-file",
+                HEALTHCHECK_OUTPUT_FILE,
+            ],
+            raiseonerr=False,
+        )
+        contents = self.master.get_file_contents(
+            HEALTHCHECK_OUTPUT_FILE, encoding="utf-8"
+        )
+        data = json.loads(contents)
+        for check in data:
+            assert check[0]["result"] == "ERROR"
+            assert check[0]["kw"]["msg"] == "sssd: not running"
+            assert check[0]["kw"]["status"] == "False"
+        self.master.run_command(["systemctl", "start", "sssd"],
+                                raiseonerr=False)
```

Comment:
Might want to drop the raiseonerr here since we do want things to blow up if sssd fails to start.